File: pramp/AbsoluteValueSort/javascript/absolute-value-sort.js
function compare(a, b) {
    if (Math.abs(a) < Math.abs(b)) return false;
    if (Math.abs(b) < Math.abs(a)) return true;
    return a > b;
  }
  
  function absSort(arr) {
      /**
      @param arr: integer[]
      @return: integer[]
      */
    for (let i=0; i<arr.length; i++) {
      for (let j=i; j<arr.length; j++) {
        const current = arr[i];
        const next = arr[j];
        let temp;
  
        if (compare(current, next)) {
            temp = arr[j];
            arr[j] = arr[i];
            arr[i] = temp;                    
        }    
      }
    }
    return arr;
  }

File: pramp/AbsoluteValueSort/swift/absolute-value-sort.swift
func compare(_ a: Int, _ b: Int) -> Bool {
  if abs(b) > abs(a) { return true }
  if abs(b) < abs(a) { return false }
  return b > a
}

func absSort(_ array: [Int]) -> [Int] {
  var numbers = array
  numbers.sort { (a, b) -> Bool in
    return compare(a,b)    
  }
  return numbers
}